Add the Pill Item 添加了药片

burin-ljx opened the pull request:
- 新增 ModFoodItems 类，统一管理食物相关物品注册
- 添加 PillItem 药片物品，支持药水效果
- 创建独立的食物创造模式标签页
- 添加食物相关的数据组件和标签
- 将原有食物物品从 ModItems 迁移到 ModFoodItems
- 更新相关配方引用到新的食物物品路径
- 添加药丸物品的 JEI 扩展支持
- 更新语言文件，添加食物物品的本地化文本
- 调整物品分组顺序，将食物标签页置于材料之后
- 添加对喷溅药水和滞留药水的支持
- 更新配方输出以包含弱化效果组件
- 优化药水列表生成逻辑，提高性能
- 修复配方显示中缺失的药水类型问题
- 增强JEI集成以正确反映所有药水变体
- 改进药水内容组件的数据传递准确性
- Resolved #2945

burin-ljx committed:
feat(food): 添加食物物品和配方系统

- 新增 ModFoodItems 类，统一管理食物相关物品注册
- 添加 PillItem 药片物品，支持药水效果
- 创建独立的食物创造模式标签页
- 添加食物相关的数据组件和标签
- 将原有食物物品从 ModItems 迁移到 ModFoodItems
- 更新相关配方引用到新的食物物品路径
- 添加药丸物品的 JEI 扩展支持
- 更新语言文件，添加食物物品的本地化文本
- 调整物品分组顺序，将食物标签页置于材料之后

burin-ljx committed:
feat(jei): 扩展药水配方支持多种类型

- 添加对喷溅药水和滞留药水的支持
- 更新配方输出以包含弱化效果组件
- 优化药水列表生成逻辑，提高性能
- 修复配方显示中缺失的药水类型问题
- 增强JEI集成以正确反映所有药水变体
- 改进药水内容组件的数据传递准确性

burin-ljx committed:
style: 修正代码样式

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/recipe/PillRecipe.java:
```diff
@@ -0,0 +1,101 @@
+        Optional<Holder<Potion>> potion = potionContents.potion();
+        return (item.is(Items.POTION)
+            || item.is(Items.SPLASH_POTION)
+            || item.is(Items.LINGERING_POTION))
```
请于 39 行 “(” 之后， 41 行 “)” 之前换行并调整缩进

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/recipe/PillRecipe.java:
```diff
@@ -0,0 +1,101 @@
+            if (
+                item.is(Items.POTION)
+                    || item.is(Items.SPLASH_POTION)
+                    || item.is(Items.LINGERING_POTION)
```
73 、 74 行缩进应与 72 行对齐

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/integration/jei/category/extension/PillRecipeExtension.java:
```diff
@@ -0,0 +1,87 @@
+            .filter((potion1) -> !potion1.is(Potions.WATER)
+                && !potion1.is(Potions.MUNDANE)
+                && !potion1.is(Potions.THICK)
+                && !potion1.is(Potions.AWKWARD)).toList();
```
请于 38 行 “(” 之后， 41 行 “)” 之前换行并调整缩进

Gu-ZT commented on src/main/java/dev/dubhe/anvilcraft/init/item/ModItemGroups.java:
```diff
@@ -38,6 +43,32 @@ public class ModItemGroups {
+                            return enabled && !potion1.is(Potions.WATER)
+                                && !potion1.is(Potions.MUNDANE)
+                                && !potion1.is(Potions.THICK)
+                                && !potion1.is(Potions.AWKWARD);
```
请于 57 行 “&&” 之前换行并调整缩进

burin-ljx committed:
style: 修正代码样式

burin-ljx committed:
fix(create):修复物品组标签移除问题

- 将错误移除的 ANVILCRAFT_FUNCTION_BLOCK 标签更正为 ANVILCRAFT_FOOD 标签
- 确保配方条件和材料需求逻辑不受影响
- 验证 Create 模组集成中的珠宝加工配方构建器功能正常

Gu-ZT committed:
style(java): 格式化代码并优化长行折行处理

- 调整 ModFoodItems 中配方注册的缩进与括号位置
- 优化 OtherLang 中本地化文本的换行格式
- 重构 PillItem 中 getDescriptionId 方法的参数折行
- 更新 PillRecipeExtension 中方法参数的换行样式
- 统一代码风格以提高可读性与维护性
